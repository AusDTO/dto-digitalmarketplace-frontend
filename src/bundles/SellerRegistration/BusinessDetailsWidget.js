import React from 'react'
import { Provider } from 'react-redux'
import RegisterComponent from '../../RegisterComponent'

// Need to make this work.
// import { BusinessDetails, createStore } from '../bundles/BusinessDetails'
import createStore from './redux/create'
import BusinessDetailsForm from './components/BusinessDetailsForm'

export const BusinessDetailsWidget = (props) => {
    const store = createStore(props)
    return (
        <Provider store={store}>
            <BusinessDetailsForm />
        </Provider>
    )
}

export default new RegisterComponent({ 'seller-business-details': BusinessDetailsWidget })
