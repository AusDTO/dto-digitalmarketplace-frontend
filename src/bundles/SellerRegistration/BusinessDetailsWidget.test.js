// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import {BusinessDetailsWidget} from './BusinessDetailsWidget';
import renderer from 'react-test-renderer';

test('BusinessDetailsWidget renders correctly', () => {
  const component = renderer.create(
    <BusinessDetailsWidget />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
