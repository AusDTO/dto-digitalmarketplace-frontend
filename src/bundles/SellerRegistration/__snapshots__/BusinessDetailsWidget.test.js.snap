exports[`test BusinessDetailsWidget renders correctly 1`] = `
<div
  className="layout">
  <header>
    <h1>
      Company details
    </h1>
  </header>
  <article
    className="content-main"
    role="main">
    <span />
    <form
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex="-1" />
    <form
      action={undefined}
      id="BusinessDetails__create"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}>
      <div
        className="field">
        <label
          htmlFor="summary">
          Company summary
        </label>
        <p
          className="hint"
          id="summary-hint">
          This will be displayed in search results
        </p>
        <div>
          <textarea
            aria-describedby="summary-hint"
            className=""
            defaultValue=""
            id="summary"
            name="summary"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]} />
          <span
            className="word-count-counter">
            50 words remaining
          </span>
        </div>
      </div>
      <div
        className="field">
        <label
          className="question-heading"
          htmlFor="website">
          Website URL
        </label>
        <input
          aria-describedby={undefined}
          className=""
          id="website"
          name="website"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="" />
      </div>
      <div
        className="field">
        <label
          className="question-heading"
          htmlFor="linkedin">
          LinkedIn URL
        </label>
        <input
          aria-describedby={undefined}
          className=""
          id="linkedin"
          name="linkedin"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="" />
      </div>
      <fieldset>
        <legend>
          Company address
        </legend>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="addressLine">
            Address
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="addressLine"
            name="address.addressLine"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="suburb">
            Suburb
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="suburb"
            name="address.suburb"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="state">
            State
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="state"
            name="address.state"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="postalCode">
            Postcode
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="postalCode"
            name="address.postalCode"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
      </fieldset>
      <input
        onClick={[Function]}
        role="button"
        type="submit"
        value="Next" />
    </form>
  </article>
</div>
`;
